desi-extract

```diff
@@ -23,20 +32,15 @@ parser.add_argument("-c", "--camera", type=str, choices=["b", "r", "z"], help="c
 args = parser.parse_args()
 
 from mpi4py import MPI
-from desispec.scripts import extract
-import sys, os, glob, socket
-import random
-random.seed(1)
```

Comment:
For consideration: the original reason for having these imports here was to let MPI wake up and establish communication before doing expensive imports like numpy/scipy/astropy that are buried in desispec.scripts.extract.  That may not matter with the docker containers, so I'm ok with moving the imports up but just wanted you to be purposeful / aware about this.

Reply:
It's all good start-up to me as far as I'm concerned.  All the Python processes wake up and start doing those imports regardless of whether `MPI_Init()` has been called.  Of course there are other imports happening down the line in the extraction code itself.

Reply:
This is in the category of black magic best practices, but we have definitely had problems in the past where numpy/astropy/scipy imports were taking so long that SLURM or MPI would kill us off or give mystery errors when we finally got to trying to connect.  Moving the MPI imports to be as soon as possible made those instabilities go away and we moved on, even if the previous "late import" of MPI should have worked.  With docker, the imports are fast enough that it shouldn't matter so OK to leave these PR changes.  Just file that away in the back of your head if you see mystery intermittent MPI failures in some other context.

Reply:
Ah that is what you mean.  To me, it suggests Shifter lets you write code the way you're supposed to in Python...